code/game/objects/items/melee/transforming.dm

```diff
@@ -99,3 +103,60 @@
 	if(force_on + increment > max)
 		return COMPONENT_BLOCK_SHARPEN_MAXED
 	sharpened_bonus = increment
+
+/obj/item/melee/transforming/butter_fly
+	name = "butterfly knife"
+	icon_state = "butterfly"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	desc = "Float like a butterfly, sting like a knife."
+	flags_1 = CONDUCT_1
+	force = 3
+	w_class = WEIGHT_CLASS_SMALL
+	throwforce = 5
+	throw_speed = 3
+	throw_range = 6
+	bare_wound_bonus = 5
+	stealthy_audio = FALSE
+	force_on = 15
+	icon_state_on = "butterfly_ext"
+	hitsound_on = 'sound/weapons/bladeslice.ogg'
+	attack_verb_on = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "cuts")
+	attack_verb_off = list("pokes")
+	w_class_on = WEIGHT_CLASS_SMALL
+	sound_when_on = 'sound/weapons/batonextend.ogg'
+	sound_when_off = 'sound/weapons/batonextend.ogg'
+	///damage dealt on backstab
+	var/backstab_damage = 40
+	COOLDOWN_DECLARE(stab_cooldown)
+	///cooldown on stabbing
+	var/stab_cooldown_time = 3 SECONDS
+
+/obj/item/melee/transforming/butter_fly/attack_alt(mob/living/victim, mob/living/user, params)
+	if(!COOLDOWN_FINISHED(src, stab_cooldown))
+		to_chat(user, "<span class='warning'>You aren't ready to backstab!</span>")
+		return ALT_ATTACK_CONTINUE_CHAIN
```

Comment:
I don't think this is getting merged anyway, but for the record--this should be CANCEL_CHAIN. You don't want to just hit the guy in the back with a not-backstab on cooldown.